### ISSUE TITLE
TreeConstructor.NodeContents.Tag.selfClosing is misleading

### ISSUE THREAD

**Issue by BlackGlory** [mykolaharmash/hyntax#31]
In the source code, this property actually represents [void elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements).

https://github.com/mykolaharmash/hyntax/blob/b09cd799498474ccf38da7f7545530062ee239ab/lib/tree-constructor-context-handlers/tag.js#L34-L50

Therefore, `selfClosing` of HTML tags like `<div />` will always be `false`.

**Comment by mykolaharmash** [comment]
Thanks, that is true, misleading indeed. There is a long-overdue issue with [SVG support](https://github.com/mykolaharmash/hyntax/issues/16), it will introduce parsing of "real" self-closing tags. Then I'll refactor and change current self-closing tags to be void elements, and rename the property in AST.

**Comment by shuerguo999** [comment]
I meet the same problem, is there any way to parse the code correctly? Anyway, thank you for providing such an excellent parser!!
````html
<div></div>
<div />
````

**Comment by mykolaharmash** [comment]
Hey @shuerguo999, thank you! 

Could you please explain your usecase? Are you parsing some extension of HTML syntax like JSX? Asking because `<div />` is not part of HTML5 standard and is not a valid syntax.

**Comment by shuerguo999** [comment]
Yes,  <div /> is not part of HTML5 standard. But I hope hyntax can support it, I use it parsing many extension of HTML syntax like 'Mini apps'、'mustache'.... I didn't find other better parser!
